src/ray/core_worker/core_worker.cc

```diff
@@ -1525,16 +1526,36 @@ Status CoreWorker::CancelTask(const ObjectID &object_id, bool force_kill) {
     return Status::Invalid("No owner found for object.");
   }
   if (obj_addr.SerializeAsString() != rpc_address_.SerializeAsString()) {
-    return direct_task_submitter_->CancelRemoteTask(object_id, obj_addr, force_kill);
+    return direct_task_submitter_->CancelRemoteTask(object_id, obj_addr, force_kill,
+                                                    recursive);
   }
 
   auto task_spec = task_manager_->GetTaskSpec(object_id.TaskId());
   if (task_spec.has_value() && !task_spec.value().IsActorCreationTask()) {
-    return direct_task_submitter_->CancelTask(task_spec.value(), force_kill);
+    return CancelTaskLocalHelper(task_spec.value(), force_kill, recursive);
   }
   return Status::OK();
 }
 
+Status CoreWorker::CancelTaskLocalHelper(const TaskSpecification &task_spec,
+                                         bool force_kill, bool recursive) {
+  auto local_cancel_result = direct_task_submitter_->CancelTask(task_spec, force_kill);
+  RAY_LOG(WARNING) << "Task " << task_spec.TaskId()
+                   << " failed to cancel:  " << local_cancel_result.message();
```

Comment:
Should this be guarded by checking if it cancelled correctly?

Reply:
I was thinking it should explicitly **not be** because the intent is best-attempt--so try to cancel as much as possible.